Automattic/wp-calypso
# Gutenboarding: add a generic event call for error

## Changes proposed in this Pull Request

This PR won't win any Github fashion awards, but it has utility, which is all we can ask. 

In short, it:

- Introduces a generic tracking event for onboarding errors. See: pbAok1-Ao-p2#comment-1260 
- Implements that generic tracking event for signup errors in the account creation step
- Adds types for function arguments properties, where those arguments are objects
- Sneaks in the `blog_id` property for the onboarding complete tracking event introduced in #41264

Event registration PR: https://github.com/Automattic/tracks-events-registration/pull/240

## Testing instructions

Head over to `/new` and run throw the flow until you get to the signup step.

Trigger an error in signup, e.g., an email address that's already registered, a password < 4 characters
<img width="558" alt="Screen Shot 2020-04-20 at 8 33 26 pm" src="https://user-images.githubusercontent.com/6458278/79742571-5e398a00-8346-11ea-98af-5f273d778429.png">

Check out `calypso_signup_error` in the network console! Are the `step` and `error` properties filled in by any chance?
<img width="471" alt="Screen Shot 2020-04-20 at 8 32 55 pm" src="https://user-images.githubusercontent.com/6458278/79742583-6265a780-8346-11ea-895a-752e402e0e30.png">

Signup and take squiz at the `calypso_signup_complete` event in the network console. `blog_id` should be filled in.

<img width="375" alt="Screen Shot 2020-04-20 at 8 36 42 pm" src="https://user-images.githubusercontent.com/6458278/79742824-c4bea800-8346-11ea-9599-3921121ab59b.png">

Fixes a bit of #41059
